src/httpClient/httpClient.ts

```diff
@@ -177,6 +189,16 @@ export default class HttpClient {
     };
   }
 
+  public static generateCacheKey(req: AxiosRequestConfig): string {
+    const prefix = req.headers?.Authorization
+      ? parseJwtCanonicalId(req.headers.Authorization.replace('Bearer ', ''))
+      : 'shared';
+    const url = `${req.baseURL ? req.baseURL : ''}${req.url}`;
+    const query = req.params ? JSON.stringify(req.params) : ''; // possible improvement: optimize cache-hit ratio by sorting the query params
+    const key = `${prefix}/${url}${query}`;
+    return req.data ? key + md5(req.data) : key;
```

Comment:
[Optional] Can we use `${}` interpolation instead of concatenation here too?

Reply:
Yes, changed to match the style of other statements.